apps/avatax/src/modules/avatax/avatax-connection.router.ts

```diff
@@ -183,11 +209,20 @@ export const avataxConnectionRouter = router({
         saleorApiUrl: ctx.saleorApiUrl,
       });
 
-      const authValidation = new AvataxEditAuthValidationService({
-        appId: ctx.appId!,
-        client: ctx.apiClient,
-        saleorApiUrl: ctx.saleorApiUrl,
-      });
+      const authValidation = new AvataxEditAuthValidationService(
+        new AvataxPatchInputTransformer(
+          new AvataxConnectionService(
+            new AvataxConnectionRepository(
+              new CrudSettingsManager(
```

Comment:
**Question:** I don't know if it will be too much of refactor but maybe this can take object instead of params? As I see that someone can mix sequence and do:

```tsx
new CrudSettingsManager(
createSettingsManager(ctx.apiClient, ctx.appId, metadataCache),
TAX_PROVIDER_KEY,
ctx.saleorApiUrl
)
```